subworkflows/local/starsolo.nf

```diff
@@ -53,14 +54,16 @@ workflow STARSOLO {
     )
     ch_versions = ch_versions.mix(STAR_ALIGN.out.versions)
 
+    // generate channel of star counts with correct metadata
+    ch_star_counts =
+        STAR_ALIGN.out.raw_counts.map{ meta, files -> [meta + [input_type: 'raw'], files] }
+        .mix( STAR_ALIGN.out.filtered_counts.map{ meta, files -> [meta + [input_type: 'filtered'], files] } )
+
 
     emit:
     ch_versions
     // get rid of meta for star index
-    star_index  = star_index.map{ meta, index -> index }
     star_result = STAR_ALIGN.out.tab
-    star_counts = STAR_ALIGN.out.counts
-    raw_counts = STAR_ALIGN.out.raw_counts
-    filtered_counts = STAR_ALIGN.out.filtered_counts
+    star_counts = ch_star_counts
```

Comment:
to me it made sense that counts and filtered counts are separate output channles in each aligner module that supports it. I rarely would want them into the same channel downstream after all.

Reply:
cool, so should do the filter in all, and not emit the collected, right?

Reply:
made the changes back, now the subworkflows are outputting them seprate, and i am mixing in the main workflow